src/Renderer/AtlasBuilder.js

```diff
@@ -0,0 +1,94 @@
+import * as THREE from 'three';
+import Capabilities from '../Core/System/Capabilities';
+import fit from './Packer';
+
+export default function pack(images, uvs) {
+    const atlasCanvas = document.createElement('canvas');
+    const colorBleedHalfOffset = (images.length == 1) ? 0 : 2;
+    const maxSize = Capabilities.getMaxTextureSize();
+    const blocks = [];
+
+    for (let i = 0; i < images.length; i++) {
+        const img = images[i];
+        const replaceWithEmpty = !img;
+        const sWidth = replaceWithEmpty ? 1 : img.width;
+        const sHeight = replaceWithEmpty ? 1 : img.height;
+
+        blocks.push({
+            index: i,
+            w: sWidth + 2 * colorBleedHalfOffset,
+            h: sHeight,
+            empty: replaceWithEmpty,
+        });
+    }
+
+    blocks.sort((a, b) => Math.max(a.w, a.h) < Math.max(b.w, b.h));
```

Comment:
Why do you need to sort? (I feel like I don't really understand your data layout)